### PR TITLE
Refactor Int type to Long type in MC stats library

### PR DESCRIPTION
**Pull request by riemanli**
## TL;DR
This pull request changes the data type of `Reach` and `Impression` variables from Int to Long in multiple functions and test cases within the code. 

## Why make this change
- The change to the data type of the "totalReach" variable from Int to Long allows for larger values to be used in the computation of variances.
- Updating the data classes to use Long data types instead of Int for certain variables ensures consistency and accuracy in the computation of variances.
- The change improves the precision and accuracy of the computations, especially when dealing with large reach values.

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1256)

**Comment by riemanli** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1226** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1226" target="_blank">Graphite</a> 
    * **PR #1233** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1233" target="_blank">Graphite</a> 
    * **PR #1184** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1184" target="_blank">Graphite</a> 
      * **PR #1247** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1247" target="_blank">Graphite</a> 
        * **PR #1252** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1252" target="_blank">Graphite</a> 
          * **PR #1256** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1256" target="_blank">Graphite</a>  👈
            * **PR #1253** <a href="https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1253" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/world-federation-of-advertisers/cross-media-measurement/1256?utm_source=stack-comment).

**Review by Marco-Premier — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1256)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @riemanli)
